src/nvim/edit.c

```diff
@@ -5278,21 +5278,23 @@ insertchar (
   } else {
     int cc;
 
+    // TODO(timeyyy): refactor  this if block away, need to manually test that 
+    // inserts still work as the tests don't catch it
```

Comment:
I will get around to it. Getting some wierd runtime errors i want to have a look at first.